src/lib/state-incentives-calculation.ts

```diff
@@ -199,9 +200,23 @@ export function calculateStateIncentivesAndSavings(
     savings[item.payment_methods[0]] += amount;
   });
 
+  // Get state tax owed to determine max potential tax savings
+  const stateTaxOwed = estimateStateTaxAmount(
+    request.household_income,
+    stateId,
+  );
+
+  // You can't save more than tax owed. Choose the lesser of state tax owed vs tax credit savings
+  const maxTaxCreditSavings = stateTaxOwed
+    ? Math.min(stateTaxOwed.taxOwed, savings.tax_credit)
+    : savings.tax_credit;
```

Comment:
I think you can just update `savings.tax_credit` in place and avoid the part down below